### PR TITLE
Replace Ubuntu -> Debian SID

### PR DESCRIPTION
**Pull request by cevich**
Ref: https://github.com/containers/automation_images/pull/250

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by cevich** [comment]
@edsantiago ping - any chance you've seen [this or a similar error (log)](https://cirrus-ci.com/task/4689002125590528) before:

```
...cut...
+ make completions  # ./contrib/cirrus/postbuild.sh:25 in main()
...cut...
# key = shell, value = completion filename
declare -A outfiles=([bash]=%s [zsh]=_%s [fish]=%s.fish [powershell]=%s.ps1);\
for shell in ${!outfiles[*]}; do \
    for remote in "" "-remote"; do \
	podman="podman$remote"; \
	outfile=$(printf "completions/$shell/${outfiles[$shell]}" $podman); \
	./bin/$podman completion $shell >| $outfile; \
    done;\
done
/bin/sh: 1: Syntax error: "(" unexpected
make: *** [Makefile:446: completions] Error 2

Exit status: 2
```

***NOTE***: This is on a brand-new Debian SID VM, it seems highly-likely I've screwed something up in the setup.

**Comment by edsantiago** [comment]
I was _just_ looking at this PR. The `/bin/sh` concerns me: it really should be `bash`. Can you think of somewhere that is using `sh` instead of `bash`?

Also, Cirrus or VMs are acting *weird* today. [example](https://api.cirrus-ci.com/v1/task/5997791034474496/logs/df.log)

**Comment by cevich** [comment]
> Can you think of somewhere that is using sh instead of bash?

When this was ubuntu, it used something called "dash" which it went to great lengths to pass-off as "bash".  I just double-checked, and this is something I removed from the debian setup.  Maybe this is to blame?

> Also, Cirrus or VMs are acting weird today.

Okay, that's proper **W-3-1-R-|)**.  Like a git-clone silently failed to obtain all the files....oh! [Look at the WebUI for that task.](https://cirrus-ci.com/task/5997791034474496) it show "skipped" for "clone", "setup" and "main".  That can't be right.  Ahh, at the top it says "PR was closed!"  So maybe this was simply a task being canceled at a strange point?

**Comment by cevich** [comment]
Maybe dash is it.  I looked in a "rerun with terminal" and found:

```
root@cirrus-task-6633121018281984:/# echo $CIRRUS_SHELL
/bin/bash
root@cirrus-task-6633121018281984:/# ls -la /bin/sh
lrwxrwxrwx 1 root root 4 Jan  5 13:20 /bin/sh -> dash
root@cirrus-task-6633121018281984:/# 
```

On Fedora, /bin/sh -> bash, so perhaps the Makefile is using "sh" but treating it as if it's really bash?

**Comment by cevich** [comment]
Ahh, [yup make defaults to /bin/sh if $SHELL is unset](https://www.gnu.org/software/make/manual/html_node/Choosing-the-Shell.html).  So that's a simple fix I think - define SHELL in our Makefile.

**Comment by cevich** [comment]
Well that's good news, only 12 test failures and the problem/solution is rather obvious:

`msg="aardvark-dns binary not found, container dns will not be enabled"`

I checked the build logs, and indeed the podman package only brings in netavark, not aardvark-dns.

**Comment by cevich** [comment]
@siretart ping - On SID, it seems installing the `podman` package doesn't bring in all necessary dependencies.  Debian SID [CI VM Image build-log](https://cirrus-ci.com/task/5019989619507200) shows it only installs netavark and not aardvark-dns.  [The script that sets up the image](https://github.com/containers/automation_images/pull/250/files#diff-727e9289c12e01646bb49f30cb044cfcdf752a05d2137b49235a1a7bcd77da8d) mentions 'podman' and assumes that will haul in everything necessary.  Is this an incorrect assumption for Debian?

The `netavark` and `aardvark-dns` packages should be basically locked-together-at-the-hip together.  They should always be installed together, with matching versions.  Podman won't work without both, or the alternative `containernetworking-plugins`.

Can you help fix this or is there some packaging policy I'm misunderstanding for Debian?

**Comment by siretart** [comment]
@cevich  I believe you found and oversight on my part, I need to add a package relationship from netavark to aardvark-dns.

Can you please help me decide whether this should be an absolute, unbreakable dependency (meaning, you can't install netavark witout aardvark-dns) or wither there are some situation where instaling netavark makes sense not having aardvark-dns installed (in which case, the relationship should be a recommends)?

**Comment by cevich** [comment]
> Can you please help me decide

Happy to, with the caveat that I'm not a packaging-guy.  IMHO, they should be locked-at-the-hip together.  Their versions need to 100% match and they should always be installed/uninstalled/upgraded together.  They can safely coexist with containernetworking-plugins, but that's a legacy solution.  For new installations, nv/av should be preferred unless the user explicitly wants containernetworking-plugins.

Does that help?

**Comment by Luap99** [comment]
@siretart 
You should be able to install netavark without aardvark-dns, aardvark is only needed when a users wants dns for their container names. Almost all people want this so it should be installed by default, I assume this maps to an recommends in packaging terms.

**Comment by cevich** [comment]
@giuseppe got a strange one here:  The new `debian-12` VM is using CgroupsV1 and runc.  The `podman play kube with disabled cgroup` test behaves counter-intuitively (to me at least).  The results are:

* [int podman debian-12 root host](https://api.cirrus-ci.com/v1/artifact/task/4559945404776448/html/int-podman-debian-12-root-host.log.html#t--podman-play-kube-with-disabled-cgroup--1) **FAILS:** `Error: requested OCI runtime runc is not compatible with NoCgroups: invalid argument`
* [int podman debian-12 rootless host](https://api.cirrus-ci.com/v1/artifact/task/5404370334908416/html/int-podman-debian-12-rootless-host.log.html#t--podman-play-kube-with-disabled-cgroup--1) **FAILS:** /same error/
* [int remote debian-12 root host](https://api.cirrus-ci.com/v1/artifact/task/6248795265040384/html/int-remote-debian-12-root-host.log.html#t--podman-play-kube-with-disabled-cgroup--1) **PASSES**

**Comment by cevich** [comment]
@containers/podman-maintainers PTAL this odd failure, seems (to me) like it shouldn't have anything to do with runc-vs-crun.  Regardless, it reproduces [on `int podman debian-12 root host`](https://api.cirrus-ci.com/v1/artifact/task/4559945404776448/html/int-podman-debian-12-root-host.log.html#t--podman-uidmapping-and-gidmapping-with-an-idmapped-volume--1) and [`int remote debian-12 root host` too](https://api.cirrus-ci.com/v1/artifact/task/6248795265040384/html/int-remote-debian-12-root-host.log.html#t--podman-uidmapping-and-gidmapping-with-an-idmapped-volume--1):

```
podman uidmapping and gidmapping with an idmapped volume
...cut...
         # podman [options] run --uidmap=0:1:500 --gidmap=0:200:5000 -v my-foo-volume:/foo:Z,idmap alpine stat -c #%u:%g# /foo
         #65534:65534#
...cut...
           Expected
               <string>: #65534:65534#
           to contain substring
               <string>: #0:0#
...cut...
```

Also (bad): I don't see any documentation for using an `idmap` option under the `--volume` man page section.  It does show up for `--mount` so maybe the test code is wrong?

**Comment by edsantiago** [comment]
Package versions:
```
Kernel:  6.1.0-3-cloud-amd64
Cgroups:  tmpfs
dpkg-query: no packages found matching containers-common
dpkg-query: no packages found matching cri-o-runc
conmon-2.1.3+ds1-1-amd64
containernetworking-plugins-1.1.1+ds1-3+b1-amd64
criu-3.17.1-2-amd64
crun-1.5+dfsg-1+b1-amd64
golang-2:1.19~1-amd64
libseccomp2-2.5.4-1+b3-amd64
podman-4.3.1+ds1-5+b1-amd64
runc-1.1.4+ds1-1+b1-amd64
skopeo-1.9.3+ds1-1-amd64
slirp4netns-1.2.0-1-amd64
```

**Comment by cevich** [comment]
Thanks Ed.  I smell on the winds of change: "Upgrade runc" :rofl:

**Comment by siretart** [comment]
> @siretart ping - On SID, it seems installing the podman package doesn't bring in all necessary dependencies. Debian SID [CI VM Image build-log](https://cirrus-ci.com/task/5019989619507200) shows it only installs netavark and not aardvark-dns. 

I've fixed this oversight in version [netavark_1.4.0-3](https://tracker.debian.org/news/1416926/accepted-netavark-140-3-source-into-unstable/), also cf. https://tracker.debian.org/netavark

Thanks for pinging me. If I miss a ping in the future, please file a bug in Debian.

**Comment by cevich** [comment]
> Thanks for pinging me. If I miss a ping in the future, please file a bug in Debian.

Will do, thanks for the quick fix and update.

**Comment by cevich** [comment]
@edsantiago I could use some help examining (or some referrals for) the debian SID system-test results.  Esp. [the `sys podman debian-12 rootless host` generate systemd](https://api.cirrus-ci.com/v1/artifact/task/4612871817003008/html/sys-podman-debian-12-rootless-host.log.html) and quadlet tests.  Those in particular seemed to pass when run as root.

The `podman buildx - basic test` failure strikes me as possibly related to https://github.com/containers/podman/issues/17433 which also failed on a stat of volumes.

FYI: I've got to rebuild the debian images to get the `systemd-container` package (provider of `machinectl`) installed, so there's going to be a delay before I can run these tests again.

Also in case it helps, the rerun-in-terminal button may be your best bet.  I'm having trouble with the `hack/get_ci_vm.sh` script on the Debian images for some unknown reason.

**Comment by edsantiago** [comment]
```
not ok 514 podman kube --network
...
Resource limits are not supported and ignored on cgroups V1 rootless systems
```
WEIRD! This is one of the RHEL8 gating failures, and I just checked, and your PR is rebased on a fixed `main`. (#17262).

The rest ... no clue. Off to do the rerun-terminal thing.

**Comment by edsantiago** [comment]
quadlet error is:
```
Error: mkdir /sys/fs/cgroup/pids/user.slice/user-2878.slice/user@2878.service/runtime: permission denied
```
@giuseppe PTAL

**Comment by rhatdan** [comment]
SELinux or UID Permissions.

**Comment by edsantiago** [comment]
This is Debian: no SELinux.
```console
$ ls -la /sys/fs/cgroup/pids/user.slice/user-2878.slice/user@2878.service
total 0
drwxr-xr-x 2 root root 0 Feb  8 20:53 .
drwxr-xr-x 4 root root 0 Feb  8 20:53 ..
-rw-r--r-- 1 root root 0 Feb  8 21:23 cgroup.clone_children
-rw-r--r-- 1 root root 0 Feb  8 20:53 cgroup.procs
-rw-r--r-- 1 root root 0 Feb  8 21:23 notify_on_release
-r--r--r-- 1 root root 0 Feb  8 21:23 pids.current
-r--r--r-- 1 root root 0 Feb  8 21:23 pids.events
-rw-r--r-- 1 root root 0 Feb  8 20:53 pids.max
-r--r--r-- 1 root root 0 Feb  8 21:23 pids.peak
-rw-r--r-- 1 root root 0 Feb  8 21:23 tasks
```
I don't see anything user-owned under pids:
```console
$ find /sys/fs/cgroup/pids \! -user root -ls
```

**Comment by edsantiago** [comment]
`buildx`: I'm not seeing a connection with the issue you linked. That's a uid issue, the buildx problem is a problem with mounts: the built container is not actually mounting the requested volumes. `st_dev` is the same everywhere. I lost my terminal window before I could get very far, but basically `stat -c %D` on anything in the container returns the same value.

**Comment by edsantiago** [comment]
>My thought would be that the mkdir is happening without being root.

Yeah - this is rootless. I don't know enough about cgroups to know if this mkdir should work (and the bug is in the /cgroup mount) or if the bug is that podman should not be trying to `mkdir`.

**Comment by rhatdan** [comment]
I was thinking that it was not happening in the user namespace.

**Comment by giuseppe** [comment]
rootless cannot use cgroupvs1:

```
Error: mkdir /sys/fs/cgroup/pids/user.slice/user-2878.slice/user@2878.service/runtime: permission denied
```

This looks like we are trying to enforce the default PID limit even for rootless on cgroupv1

**Comment by edsantiago** [comment]
>rootless cannot use cgroupvs1

Thank you. I will assume that someone else on this thread can translate that into an actionable task and a way to fix tests.

**Comment by cevich** [comment]
Update per further discussion:  I'll open an issue on the rootless CGv1 quadlet breakage and mark the tests to be skipped until a fix can be worked in.

File: test/system/700-play.bats [hunk 1]
```diff
@@ -240,8 +240,14 @@ EOF
     TESTDIR=$PODMAN_TMPDIR/testdir
     mkdir -p $TESTDIR
     echo "$testYaml" | sed "s|TESTDIR|${TESTDIR}|g" > $PODMAN_TMPDIR/test.yaml
+
+    cgv1_preface=""
+    if is_cgroupsv1 && is_rootless; then
+      cgv1_preface="Resource limits.+ignored on.+"
+    fi
+
```
Comment (cevich) [inline test/system/700-play.bats:248]: @edsantiago what did I do wrong here?  [It still fails with this prefix in-place (log)](https://api.cirrus-ci.com/v1/artifact/task/5358200241782784/html/sys-podman-debian-12-rootless-host.log.html).

**Comment by cevich** [comment]
Note to self: [Opened an issue for the rootless buildx test failure.](https://github.com/containers/podman/issues/17520)

File: test/system/700-play.bats (hunk 1 above)
Comment (edsantiago) [reply test/system/700-play.bats:248]: Unfortunately, and counterintuitively, `assert` uses `expr`. You have to backslash the pluses... or just use dot-star.

Comment (cevich) [reply test/system/700-play.bats:248]: Ahh thanks, that would have taken me a long while to figure out.

**Comment by edsantiago** [comment]
```
$ podman pod inspect --format {{.InfraContainerID}} Pod:
Error: inspecting object: no such pod Pod:
```
Cause: tests expect that `podman kube play` will emit a pod ID. The nasty warning interferes with all output.

I suggest skipping this test on Debian until someone fixes the nasty warning.

**Comment by cevich** [comment]
Thanks Ed.  By "nasty warning" do you mean:

`Resource limits are not supported and ignored on cgroups V1 rootless systems`

I believe I dealt with this same message on another test and Giuseppe said "nope, can't work".  I'll try and add this to that issue if I can find it, if not I'll open another one.

**Comment by cevich** [comment]
I found it.  It's the nasty warning bypassed in the `Fix 'podman kube --network' test for rootless CGv1` test, there's no open issue that I can find.  I'll open one.

**Comment by edsantiago** [comment]
The buildah-bud failure is a [hard failure, not a flake](https://github.com/containers/buildah/pull/4396) and has been disabled in buildah-land. It needs to be disabled here too.  Patch: [disable-failing-bud-test.patch.txt](https://github.com/containers/podman/files/10787827/disable-failing-bud-test.patch.txt)


```diff
diff --git a/test/buildah-bud/apply-podman-deltas b/test/buildah-bud/apply-podman-deltas
index 06d69ed03..0d3aef158 100755
--- a/test/buildah-bud/apply-podman-deltas
+++ b/test/buildah-bud/apply-podman-deltas
@@ -249,5 +249,12 @@ skip_if_remote "different error messages between podman & podman-remote" \
 
 # END   tests which are skipped due to actual podman or podman-remote bugs.
 ###############################################################################
+# BEGIN temporary workarounds that must be reevaluated periodically
+
+skip "Test is completely broken, see buildah issue 4396" \
+     "bud-multiple-platform-values"
+
+# END   temporary workarounds that must be reevaluated periodically
+###############################################################################
 
 exit $RC
```

**Comment by cevich** [comment]
Thanks @edsantiago

File: test/system/070-build.bats [hunk 2]
```diff
@@ -30,6 +30,11 @@ EOF
 }
 
 @test "podman buildx - basic test" {
+    driver=$(run_podman info --format "{{.Store.GraphDriverName}}")
```
Comment (edsantiago) [inline test/system/070-build.bats:33]: This does not work. One cannot use `run_podman` this way. You need to do:
```bash
    run_podman info blah blah
    if [[ "$output" == "vfs" ]]; then ...skip
```

Comment (cevich) [reply test/system/070-build.bats:33]: ahh ok, I'll fix it.

**Comment by cevich** [comment]
Force-push: Rebased, corrected buildx skip, updated VM images to https://github.com/containers/automation_images/pull/256

**Comment by edsantiago** [comment]
Two test failures are flakes, one well known (#17216, currently being ignored), the other a new one (filed #17598).

@containers/podman-maintainers PTAL, let's get this in.

/lgtm
/hold

New CI map: cirrus-map-pr17305

**Comment by edsantiago** [comment]
Sigh. pasta tests failing repeatedly despite multiple reruns. @cevich I think you'll need to rebuild VMs and downgrade pasta: older version fails, but it's a flake; this version is a hard fail. Issue being tracked in #17598.

**Comment by edsantiago** [comment]
@cevich there's a [new pasta build in progress](https://github.com/containers/podman/issues/17598#issuecomment-1439990574).

There are also at least two in-flight PRs that I am blocking from merging, because they are runc-related and I don't want more CI disasters. 

Building new VMs with new pasta will take time and (sigh) introduce new surprise breakages, maybe pasta, maybe some other package.

PROPOSAL: just skip all the pasta tests in this PR (or maybe just the right subset, if you can confidently identify those). Get this to pass, merge it, then do pasta cleanup in a followup PR. That will allow the stuck runc PRs to proceed, and will give time to debug new VMs.

**Comment by cevich** [comment]
Force-push: Rebased, resolved merge conflict, re-worded temp. -> perm. test skip, added skip for all pasta tests.

**Comment by cevich** [comment]
[Restarted `sys podman debian-12 root host` task - autoupdate test failure](https://api.cirrus-ci.com/v1/artifact/task/6064714845782016/html/sys-podman-debian-12-root-host.log.html) (assuming a flake).

**Comment by cevich** [comment]
@containers/podman-maintainers CI is Green!

**Pull request merged**